README.md
# shared_store

Native storage solution based on MMKV.

##  Example
#### iOS 
<div align=left> 
<img width="180" height="330" src="https://user-images.githubusercontent.com/68054922/156133619-de4c9235-aef0-452b-aa52-3c0552429392.png"/>
<img width="180" height="330" src="https://user-images.githubusercontent.com/68054922/156133622-bd4c3462-a8e8-45fc-9495-6fdada7b3c74.png"/>
<img width="180" height="330" src="https://user-images.githubusercontent.com/68054922/156133629-dcccb683-be33-457c-bb26-0fe2b8f8e748.png"/>
<img width="180" height="330" src="https://user-images.githubusercontent.com/68054922/156133634-69fc35c6-0123-4afb-8133-5440e5fb655a.png"/>
<img width="180" height="330" src="https://user-images.githubusercontent.com/68054922/156133639-ce04bfad-c852-435e-b7b1-7ce0f9c451c9.png"/>
</div>

#### Android
<div align=left> 
<img width="180" height="330" src="https://user-images.githubusercontent.com/68054922/156137697-08088572-5224-481a-89d4-2abb184cb9c7.png"/>
<img width="180" height="330" src="https://user-images.githubusercontent.com/68054922/156137705-f1b26b31-3267-41a9-9fed-8ee98f2f0648.png"/>
<img width="180" height="330" src="https://user-images.githubusercontent.com/68054922/156137708-acc515d0-47a0-43b1-b0e9-1290ff770b44.png"/>
<img width="180" height="330" src="https://user-images.githubusercontent.com/68054922/156137713-e134eb39-c727-477c-90ac-36f09a08afc9.png"/>
<img width="180" height="330" src="https://user-images.githubusercontent.com/68054922/156137717-4bfe4d12-0c73-4979-b6a4-eab8e9831636.png"/>
</div>


## Usage
### 一,Flutter端
1.添加依赖：

```dart
dependencies:
  flutter:
    sdk: flutter

  shared_store: ^0.01
```

2.启用SharedStore前，导入`shared_store_plugin.dart`对其进行初始化：

```dart
import 'package:shared_store/shared_store_plugin.dart';
```

```dart
SharedStorePlugin.initMMKV();
```

3.根据是否需要添加除内置默认的`default`MMKV外的可选的MMKV实例：
```dart
    String customMMKVId = "testPlugin";
    SharedStorePlugin.addMMKV(customMMKVId);
```

4.在需要存储数据时：
```dart
    const bool testBool = true;
    const int testInt = 123456;
    const double testDouble = 3.1415926;
    const String testString = 'testStringValue';
    String? boolResult = await SharedStorePlugin.storeBool('testBool', testBool);
    String? doubleResult = await SharedStorePlugin.storeDouble('testDouble', testDouble);
    String? intResult = await SharedStorePlugin.storeInt('testInt', testInt);
    String? stringResult = await SharedStorePlugin.storeString('testString', testString);
```

5.在需要读取数据时：
```dart
    bool? readBool = await SharedStorePlugin.readBool('testBool');
    int? readInt = await SharedStorePlugin.readInt('testInt');
    double? readDouble = await SharedStorePlugin.readDouble('testDouble');
    String? readString = await SharedStorePlugin.readString('testString');
```
6.在需要删除存储数据时：
```dart
SharedStorePlugin.removeValue('testBool');
```


CHANGELOG.md
## 0.0.1

* Implement Flutter call native to store data


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:shared_store/shared_store_plugin.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatefulWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  late String result = '';
  @override
  void initState() {
    super.initState();
  }

  void initMMKV() {
    SharedStorePlugin.initMMKV();
    setState(() {
      result = 'init success';
    });
  }

  void addMMKV() {
    String customMMKVId = "testPlugin";
    SharedStorePlugin.addMMKV(customMMKVId);
    setState(() {
      result = "MMKV $customMMKVId add success";
    });
  }

  void storeTestData() async {
    const bool testBool = true;
    const int testInt = 123456;
    const double testDouble = 3.1415926;
    const String testString = 'testStringValue';
    String? boolResult = await SharedStorePlugin.storeBool('testBool', testBool);
    String? doubleResult = await SharedStorePlugin.storeDouble('testDouble', testDouble);
    String? intResult = await SharedStorePlugin.storeInt('testInt', testInt);
    String? stringResult = await SharedStorePlugin.storeString('testString', testString);
    setState(() {
      if (boolResult == 'true' && doubleResult == 'true' && intResult == 'true' && stringResult == 'true') {
        result =
            'store value : \n boolValue: $testBool \n intValue: $testInt \n doubleValue: $testDouble \n stringValue: $testString \n store value success';
      } else {
        result = 'store value failed';
      }
    });
  }

  void readTestData() async {
    bool? readBool = await SharedStorePlugin.readBool('testBool');
    int? readInt = await SharedStorePlugin.readInt('testInt');
    double? readDouble = await SharedStorePlugin.readDouble('testDouble');
    String? readString = await SharedStorePlugin.readString('testString');
    setState(() {
      result =
          'read value : \n boolValue: $readBool \n intValue: $readInt \n doubleValue: $readDouble \n stringValue: $readString \n read value success';
    });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('SharedStore example app'),
        ),
        body: Padding(
          padding: const EdgeInsets.all(20.0),
          child: Column(
            children: [
              Row(
                children: [
                  CustomButton(
                      title: '1,initMMKV',
                      onTap: () {
                        initMMKV();
                      }),
                  Expanded(
                    child: Container(),
                  ),
                  CustomButton(
                      title: '2,addMMKV',
                      onTap: () {
                        addMMKV();
                      })
                ],
              ),
              const Divider(
                height: 40,
                color: Colors.white,
              ),
              Row(
                children: [
                  CustomButton(
                      title: '3,setData',
                      onTap: () {
                        storeTestData();
                      }),
                  Expanded(child: Container()),
                  CustomButton(
                      title: '4,getData',
                      onTap: () {
                        readTestData();
                      })
                ],
              ),
              const Divider(
                height: 20,
                color: Colors.white,
              ),
              ConsolePage(consoleText: result),
            ],
          ),
        ),
      ),
    );
  }
}

class CustomButton extends StatelessWidget {
  const CustomButton({Key? key, required this.title, required this.onTap}) : super(key: key);
  final String title;
  final void Function() onTap;
  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        decoration: BoxDecoration(
            color: Colors.white,
            border: Border.all(color: Colors.black, width: 0.5),
            borderRadius: const BorderRadius.all(Radius.circular(5)),
            boxShadow: const [
              BoxShadow(color: Colors.black, offset: Offset(3.0, 3.0), blurRadius: 5.0, spreadRadius: 1.0),
              BoxShadow(color: Colors.grey, offset: Offset(-3.0, -3.0), blurRadius: 5.0, spreadRadius: 1.0)
            ]),
        child: Padding(
          padding: const EdgeInsets.all(8.0),
          child: Center(child: Text(title)),
        ),
      ),
    );
  }
}

class ConsolePage extends StatelessWidget {
  const ConsolePage({Key? key, this.consoleText = ""}) : super(key: key);
  final String consoleText;
  @override
  Widget build(BuildContext context) {
    return Container(
      decoration: BoxDecoration(
        color: Colors.grey,
        border: Border.all(color: Colors.black, width: 0.5),
        borderRadius: const BorderRadius.all(Radius.circular(10)),
      ),
      child: Column(
        mainAxisAlignment: MainAxisAlignment.start,
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          const Padding(
            padding: EdgeInsets.all(5),
            child: Text('console:'),
          ),
          Padding(padding: const EdgeInsets.all(10), child: Text(consoleText)),
        ],
      ),
    );
  }
}
